modules/cli/src/main/scala/scala/cli/packaging/NativeImage.scala

```diff
@@ -231,46 +234,77 @@ object NativeImage {
         maybeWithManifestClassPath(
           createManifest = Properties.isWin,
           classPath = originalClasspath.map(os.Path(_, os.pwd))
-        ) { classPath =>
-          val args = extraOptions ++ Seq(
-            s"-H:Path=${dest / os.up}",
-            s"-H:Name=${dest.last.stripSuffix(".exe")}", // FIXME Case-insensitive strip suffix?
-            "-cp",
-            classPath.map(_.toString).mkString(File.pathSeparator),
-            mainClass
-          )
+        ) { processedClasspath =>
+          val (classPath, toClean, scala3extraOptions) =
+            if (!build.scalaParams.scalaBinaryVersion.startsWith("3"))
+              (processedClasspath, Seq[os.Path](), Seq[String]())
+            else {
+              val coursierCacheLocation = os.Path(FileCache().location.toPath())
```

Comment:
Same as above, the coursier cache isn't really meant for that… The graal processor should just compute its own default cache path with the directories-jvm library.